src/Storages/System/StorageSystemZooKeeperInfo.cpp

```diff
@@ -0,0 +1,392 @@
+        res_columns[29]->insert(stoi(dirs_responses_map["log_dir_size"]));
+
+
+        /// lgif command
+        String lgif_output = sendFourLetterCommand(host, port, "lgif");
+        std::map <String, String> lgif_responses_map = getTokens(lgif_output, '\t');
+
+        // /* 30 */ {"first_log_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper first log index."},
+        res_columns[30]->insert(stoi(lgif_responses_map["first_log_idx"]));
+
+        // /* 31 */ {"first_log_term", std::make_shared<DataTypeUInt64>(), "The ZooKeeper first log term."},
+        res_columns[31]->insert(stoi(lgif_responses_map["first_log_term"]));
+
+        // /* 32 */ {"last_log_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper last log index."},
+        res_columns[32]->insert(stoi(lgif_responses_map["last_log_idx"]));
+
+        // /* 33 */ {"last_log_term", std::make_shared<DataTypeUInt64>(), "The ZooKeeper last log term."},
+        res_columns[33]->insert(stoi(lgif_responses_map["last_log_term"]));
+
+        // /* 34 */ {"last_committed_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper last committed index."},
+        res_columns[34]->insert(stoi(lgif_responses_map["last_committed_log_idx"]));
+
+        // /* 35 */ {"leader_committed_log_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper leader committed log index."},
+        res_columns[35]->insert(stoi(lgif_responses_map["leader_committed_log_idx"]));
+
+        // /* 36 */ {"target_committed_log_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper target committed log index."},
+        res_columns[36]->insert(stoi(lgif_responses_map["target_committed_log_idx"]));
+
+        // /* 37 */ {"last_snapshot_idx", std::make_shared<DataTypeUInt64>(), "The ZooKeeper last snapshot index."},
+        res_columns[37]->insert(stoi(lgif_responses_map["last_snapshot_idx"]));
+
+    }
+
+}
+
+
+String StorageSystemZooKeeperInfo::sendFourLetterCommand(String host, String port, String command) const
+{
+    Poco::Net::SocketAddress address(host, port);
+    Poco::Net::StreamSocket socket;
+
+    size_t MAX_RESPONSE_SIZE = 1000000;
+    String response;
+        try
+        {
+            socket = Poco::Net::StreamSocket();
+            socket.connect(address);
+
+            socket.setNoDelay(true);
+
+            auto in = std::make_shared<ReadBufferFromPocoSocket>(socket);
+            auto out = std::make_shared<AutoCanceledWriteBuffer<WriteBufferFromPocoSocket>>(socket);
+
+            int32_t res = *reinterpret_cast<const int32_t *>(command.data());
+            /// keep consistent with Coordination::read method by changing big endian to little endian.
+            int32_t cmd_int = std::byteswap(res);
+
+            Coordination::write(cmd_int,*out);
+            out->next();
+
+            char ch;
+            while (in->read(ch) && response.size() <= MAX_RESPONSE_SIZE)
+                response += ch;
+
+        }
+        catch (...)
```

Comment:
Not sure about error handling. Let's make columns `Nullable(xxx)` in table. Also let's return `std::expected<String, String>` (find examples in code) and for commands which failed let's fill NULL into corresponding columns.

Reply:
addressed it